docs/src/main/asciidoc/security.adoc

```diff
@@ -3,161 +3,185 @@ This guide is maintained in the main Quarkus repository
+Quarkus Security is a framework that provides the architecture, multiple authentication and authorization mechanisms, and other tools for you to build secure and production-quality Java applications.
 
-This document provides a brief overview of Quarkus Security and links to the individual guides.
+== Security architecture
 
-== Getting Started
+The `HttpAuthenticationMechanism` interface is the main entry mechanism for securing HTTP applications in Quarkus. 
 
-Please see the xref:security-getting-started.adoc[Getting Started With Security] guide for a quick walkthrough through Quarkus Security where you can learn how to use xref:security-basic-auth-concept.adoc[Basic HTTP Authentication] mechanism and `JPA Identity Provider` to create `SecurityIdentity` and authorize a secure access to the endpoint with `Role Based Access Control`.
+The Quarkus Security Manager uses `HttpAuthenticationMechanism` to extract the authentication credentials from the HTTP request and delegates them to `IdentityProvider` to convert the credentials to `SecurityIdentity`.
+For example, the credentials can come from the `Authorization` header, client HTTPS certificates, or cookies.
 
-== Architecture
-
-`HttpAuthenticationMechanism` is the main entry into Quarkus HTTP Security.
-
-Quarkus Security Manager uses `HttpAuthenticationMechanism` to extract the authentication credentials from the HTTP request and delegates to `IdentityProvider` to
-complete the conversion of these credentials to `SecurityIdentity`.
-
-For example, the credentials may be coming with the HTTP `Authorization` header, client HTTPS certificates or cookies.
-
-`IdentityProvider` verifies the authentication credentials and maps them to `SecurityIdentity` which contains the username, roles, the original authentication credentials, and other attributes.
+`IdentityProvider` verifies the authentication credentials and maps them to `SecurityIdentity`, which has the username, roles, original authentication credentials, and other attributes.
 
 For every authenticated resource, you can inject a `SecurityIdentity` instance to get the authenticated identity information.
 
-In some other contexts you may have other parallel representations of the same information (or parts of it) such as `SecurityContext`
-for JAX-RS or `JsonWebToken` for JWT.
+In other contexts, it is possible to have other parallel representations of the same information or parts of it, for example, `SecurityContext`
+for JAX-RS or `JsonWebToken` for JSON Web Tokens (JWT).
 
-== Authentication mechanisms
+== Getting started with Quarkus Security
 
-Quarkus supports several sources to load authentication information from.
+Before you start building security into your Quarkus applications, review the overview information to learn about the different authentication and authorization mechanisms that Quarkus supports.
```

Comment:
@michelle-purcell I'm not sure about this particular refactoring - I think here we need to link to that getting started guide without interposing the info about the mechanisms

Reply:
Never mind this one :-), I see this section is still there